### PR TITLE
Real time component added

### PR DESCRIPTION
**Pull request by lukaszgancarz98**

### CODE REVIEW

File: django_react_proj/users-fe/src/components/RealTime/RealTime.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import moment from 'moment'
```
Comment (bakowroc) [inline django_react_proj/users-fe/src/components/RealTime/RealTime.js:2]: Moment.js is a deprecated library not really in maintenance. It's also really heavy compared to other libraries (and your needs) It's better to use `date-fns` or `dayjs`.

File: django_react_proj/users-fe/src/components/RealTime/RealTime.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  };
+  
+const Time = () => {
+    const [time, setTime] = React.useState(0);
+    const currentCallback = useCurrentCallback(() => {
+    const date = new Date();
+    setTime(date.toISOString());
+    });
+    React.useEffect(() => {
+    const handle = setInterval(currentCallback, 100);
```
Comment (bakowroc) [inline django_react_proj/users-fe/src/components/RealTime/RealTime.js:20]: If you want to show time up to seconds 100ms scrap time is too much. Use a full second, which is 1000ms

File: django_react_proj/users-fe/src/components/RealTime/RealTime.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    const reference = React.useRef();
+    reference.current = callback;
+    return (...args) => {
+      return reference.current?.(...args);
+    };
+  };
+  
+const Time = () => {
+    const [time, setTime] = React.useState(0);
+    const currentCallback = useCurrentCallback(() => {
```
Comment (bakowroc) [inline django_react_proj/users-fe/src/components/RealTime/RealTime.js:15]: I feel like this abstraction is unnecessary for what you want to achieve, can you explain what is it for and what this component intends to do (I might be missing something)?

File: django_react_proj/users/migrations/0001_initial.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 3.2 on 2021-04-18 14:56
+# Generated by Django 3.1.7 on 2021-04-18 20:14
```
Comment (bakowroc) [inline django_react_proj/users/migrations/0001_initial.py:1]: This looks like a problem with versioning, this might not be case right now but it's a good practice to keep hardcoded versions of packages in your `requirements.txt` you use to avoid compatibility differences.

File: django_react_proj/users/migrations/0001_initial.py [hunk 5]
```diff
@@ -20,6 +20,7 @@ class Migration(migrations.Migration):
                 ('deadline', models.TimeField(default='15:30', verbose_name='Deadline')),
                 ('done', models.BooleanField(default=False, verbose_name='Done')),
                 ('registrationDate', models.DateField(auto_now_add=True, verbose_name='Registration Date')),
+                ('email', models.CharField(max_length=254, null=True, verbose_name='Email')),
```
Comment (bakowroc) [inline django_react_proj/users/migrations/0001_initial.py:23]: This like kinda completely unrelated change to what this PR is for

File: django_react_proj/users-fe/package.json [hunk 6]
```diff
@@ -14,6 +14,8 @@
     "axios": "^0.21.1",
     "bootstrap": "^4.6.0",
     "date-fns": "^2.21.1",
+    "emailjs-com": "^2.6.4",
```
Comment (bakowroc) [inline django_react_proj/users-fe/package.json:17]: Is this package in use?

**Review by bakowroc — CHANGES_REQUESTED** [review]
Make sure you are rebased to the ucrrent `Develop` because you missing some important changes (like travis checks)

File: django_react_proj/users-fe/package.json (hunk 6 above)
Comment (lukaszgancarz98) [reply django_react_proj/users-fe/package.json:17]: Not now but in future

File: django_react_proj/users/migrations/0001_initial.py (hunk 5 above)
Comment (lukaszgancarz98) [reply django_react_proj/users/migrations/0001_initial.py:23]: Yes i know i try earlier to add functionality to send email when current time is close to deadline

File: django_react_proj/users-fe/src/components/RealTime/RealTime.js (hunk 3 above)
Comment (lukaszgancarz98) [reply django_react_proj/users-fe/src/components/RealTime/RealTime.js:15]: Now i looked on the code and i saw it could be done in more easier way without useCurrentCallBack

File: django_react_proj/users-fe/src/components/RealTime/RealTime.js (hunk 2 above)
Comment (lukaszgancarz98) [reply django_react_proj/users-fe/src/components/RealTime/RealTime.js:20]: Changed

File: django_react_proj/users-fe/src/components/RealTime/RealTime.js (hunk 1 above)
Comment (lukaszgancarz98) [reply django_react_proj/users-fe/src/components/RealTime/RealTime.js:2]: Changed to date-fns

File: django_react_proj/users/models.py [hunk 7]
```diff
@@ -6,6 +6,7 @@ class User(models.Model):
     deadline = models.TimeField('Deadline', auto_now=False, auto_now_add=False, default='15:30')
     done = models.BooleanField('Done', default=False)
     registrationDate = models.DateField("Registration Date", auto_now_add=True)
+    email = models.CharField("Email", max_length= 254, null = True)
```
Comment (karolszymonczyk) [inline django_react_proj/users/models.py:9]: it seems you don't have python prettier (different spaces around =)

File: django_react_proj/users-fe/package.json [hunk 8]
```diff
@@ -14,6 +14,8 @@
...
+    "moment": "^2.29.1",
```
Comment (karolszymonczyk) [inline django_react_proj/users-fe/package.json:18]: unused dependency since you use `date-fns`

**Review by karolszymonczyk — APPROVED** [review]

**Review by bakowroc — APPROVED** [review]
